linkerd/app/test/src/http_util.rs

```diff
@@ -1,43 +1,19 @@
 use crate::{
-    app_core::{svc, tls, Error},
+    app_core::{svc, Error},
     io, ContextError,
 };
 use futures::FutureExt;
 use hyper::{body::HttpBody, Body, Request, Response};
-use parking_lot::Mutex;
-use std::{future::Future, sync::Arc};
+use std::future::Future;
 use tokio::task::JoinHandle;
 use tower::{util::ServiceExt, Service};
 use tracing::Instrument;
 
 #[allow(deprecated)] // linkerd/linkerd2#8733
 use hyper::client::conn::{Builder as ClientBuilder, SendRequest};
 
-pub struct Server {
-    #[allow(deprecated)] // linkerd/linkerd2#8733
```

Comment:
this is cherry-picked out of a draft branch i've been working on, addressing deprecated client connection interfaces. this type isn't used, so we can save ourselves the trouble and remove this type altogether.